### PR TITLE
added cond_exprt

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: c4810ef0fc6ea479ddc17f15c6626667488ff1eb).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89423884

File: src/util/std_expr.h [hunk 1]
```diff
@@ -4750,4 +4750,49 @@ inline void validate_expr(const popcount_exprt &value)
   validate_operands(value, 1, "popcount must have one operand");
 }
 
+/// this is a parametric version of an if-expression: it returns
+/// the value of the first case (using the ordering of the operands)
+/// whose condition evaluates to true.
```
Comment (smowton) [inline src/util/std_expr.h:4755]: No else case? It's undefined if none of the conditions pass?

File: src/util/std_expr.h [hunk 2]
```diff
@@ -4750,4 +4750,49 @@ inline void validate_expr(const popcount_exprt &value)
...
+class cond_exprt : public multi_ary_exprt
```
Comment (smowton) [inline src/util/std_expr.h:4756]: Seems easily confused with `if_exprt`, how about something like `nary_if_exprt` or `if_elseif_exprt`?

Comment (kroening) [reply src/util/std_expr.h:4756]: Agreed that something like that would be better; however, ID_cond is the existing name, and renaming should be a separate job. The added class simply documents the current state.

File: src/util/std_expr.h (hunk 1 above)
Comment (kroening) [reply src/util/std_expr.h:4755]: Yes, this is bugging me as well.
The origin of this expression is SMV, where it is common to end with a 'true' case.
If you don't, you get non-determinism.

**Review by smowton — APPROVED** [review]

**Review by martin-cs — APPROVED** [review]
I'm assuming there is something that actually creates and uses these and has test cases?  Including the non-deterministic behaviour of them?

**Pull request merged**